src/components/NostoSection/NostoSection.ts

```diff
@@ -0,0 +1,56 @@
+import { nostojs } from "@nosto/nosto-js"
+import { customElement } from "../decorators"
+import { NostoElement } from "../NostoElement"
+import { addRequest } from "../NostoCampaign/orchestrator"
+import { JSONResult } from "@nosto/nosto-js/client"
+
+@customElement("nosto-section")
+export class NostoSection extends NostoElement {
+  /** @private */
+  static attributes = {
+    placement: String,
+    section: String
+  }
+
+  placement!: string
+  section!: string
+
+  async connectedCallback() {
+    this.toggleAttribute("loading", true)
+    try {
+      const api = await new Promise(nostojs)
+      const rec = (await addRequest({
+        placement: this.placement,
+        responseMode: "JSON_ORIGINAL" // TODO use a responseMode that returns only the needed data
+      })) as JSONResult
+      if (rec) {
+        const markup = await getSectionMarkup(this, rec)
+        this.innerHTML = markup
+        api.attributeProductClicksInCampaign(this, rec)
+      }
+    } finally {
+      this.toggleAttribute("loading", false)
+    }
+  }
```

Comment:
Tiny stylistic suggestion:

```suggestion
  async connectedCallback() {
    this.toggleAttribute("loading", true)
    try {
      initializeMarkup();
    } finally {
      this.toggleAttribute("loading", false)
    }
  }
  
  async initializeMarkup() {
    const api = await new Promise(nostojs)
    const rec = (await addRequest({
      placement: this.placement,
      responseMode: "JSON_ORIGINAL" // TODO use a responseMode that returns only the needed data
    })) as JSONResult
    if (!rec) {
      return
    }
    const markup = await getSectionMarkup(this, rec)
    this.innerHTML = markup
    api.attributeProductClicksInCampaign(this, rec)
  }
```